include/mfmg/common/amge.templates.hpp

```diff
@@ -231,6 +265,44 @@ void AMGe<dim, VectorType>::compute_restriction_sparse_matrix(
   restriction_sparse_matrix.compress(dealii::VectorOperation::add);
 
 #if MFMG_DEBUG
+  // Check that the locally_relevant_global_diag is the sum of the agglomerates
+  // diagonal
+  // TODO do not ask user for the locally_relevant_global_diag
+  pos = 0;
+  dealii::LinearAlgebra::distributed::Vector<typename VectorType::value_type>
+      new_global_diag(locally_relevant_global_diag.get_partitioner());
+  for (unsigned int i = 0; i < n_agglomerates; ++i)
+  {
+    unsigned int const n_local_eig = n_local_eigenvectors[i];
```

Comment:
Could you avoid using `n_local_eigenvetors` here and `eigenvectors` in the loop below? It makes it seem as if you are using some eigenvectors when computing the diagonal, when in fact you just get the right sizes out. Would be easier to read that if you just get those sizes out some other way.